olive/cli/auto_opt.py

```diff
@@ -0,0 +1,230 @@
+            "auto-opt",
+            help=("Automatically performance optimize input model"),
+        )
+
+        # model options
+        model_group = sub_parser.add_argument_group("model options")
+        model_group.add_argument(
+            "--model",
+            required=True,
+            help="Onnx input model path.",
+        )
+        model_group.add_argument(
+            "--task",
+            type=str,
+            default=None,
+            help="The task of the input model. Need to be specified if the input model is downloaded from huggingface.",
+        )
+        model_group.add_argument(
+            "--model_type",
+            type=str,
+            default="HfModel",
+            choices=["HfModel", "ONNXModel"],
+            help="model type, choose from HfModel and ONNXModel",
+        )
+
+        system_group = sub_parser.add_argument_group("system options")
+        system_group.add_argument(
+            "--device",
+            type=str,
+            default=None,
+            choices=["cpu", "gpu"],
+            # TODO(anyone): add more devices cpu_spr, npu, vpu, intel_myriad
+            help=(
+                "Device to use for optimization, choose from cpu and gpu. If not specified,"
+                " will deduce from the value of execution providers, CPU/VistisAi for cpu and"
+                " CUDA/Tensorrt/Dml for gpu. If both device and execution providers are not specified,"
+                " default to cpu device with CPUExecutionProvider."
+            ),
+        )
+        system_group.add_argument(
+            "--providers",
+            type=str,
+            nargs="*",
+            choices=["CPU", "CUDA", "Tensorrt", "Dml", "VitisAI"],
+            help="List of execution providers to use for optimization",
+        )
+
+        # dataset options
+        dataset_group = sub_parser.add_argument_group(
+            "dataset options, required for some optimization passes like quantization, and evaluation components"
+        )
+        dataset_group.add_argument(
+            "--data_config_path",
+            type=str,
+            required=True,
+            help="Path to the data config file. It allows to customize the data config(json/yaml) for the model.",
+        )
+
+        auto_opt_config_group = sub_parser.add_argument_group("auto optimizer options")
+        auto_opt_config_group.add_argument(
+            "--precisions",
+            type=str,
+            nargs="*",
+            choices=["fp16", "fp32", "int4", "int8"],
+            help=(
+                "The output precision of the optimized model. If not specified, "
+                "the default precision is fp32 for cpu and fp16 for gpu"
+            ),
+        )
+
+        search_strategy_group = sub_parser.add_argument_group("search strategy options")
+        search_strategy_group.add_argument(
+            "--num-samples", type=int, default=5, help="Number of samples for search algorithm"
```

Comment:
Do we really need to expose this in the CLI?

Reply:
I was thinking if the search takes long time, user can somehow reduce the numbers of num-samples to stop the search in time.